### PR TITLE
Add tests for dumping window views, and fix dumping of COUNT(*) OVER (...)

### PR DESCRIPTION
**Pull request by hlinnaka**

### CODE REVIEW

**Comment by vraghavan78** [comment]
do these tests work with GPORCA?

**Comment by hlinnaka** [comment]
On 09/07/2017 06:23 AM, Venkatesh Raghavan wrote:
> do these tests work with GPORCA?

Should work. Deparsing works on the parse tree, before planning.

- Heikki

File: src/test/regress/sql/window_views.sql [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  select avg(a) over (w)
+  from testtab window w as (order by a rows between 1 preceding and 1 following);
+select pg_get_viewdef('v9');
+
+create view v10 as
+  select avg(a) over (w rows between 1 preceding and 1 following)
+  from testtab window w as (order by a);
+select pg_get_viewdef('v10');
+
+-- This is not allowed in PostgreSQL. Does it actually work on GPDB?
```
Comment (danielgustafsson) [inline src/test/regress/sql/window_views.sql]: Would be nice with an expanded comment stating exactly what isn't allowed in PostgreSQL.

**Comment by hlinnaka** [comment]
Expanded the comment a bit, and pushed. Thanks!

**Comment by bhuvnesh2703** [comment]
@hlinnaka should this be ported to 5X_STABLE as well?

**Comment by hlinnaka** [comment]
On 09/12/2017 01:27 AM, Bhuvnesh wrote:
> @hlinnaka should this be ported to 5X_STABLE as well?

Can't, unfortunately. It adds fields to WindowRef, which is a catalog 
change.

- Heikki

**Pull request merged**